core/promise/consumer.go

```diff
@@ -36,17 +66,49 @@ func (c *Consumer) NewRequest() (requestPtr interface{}) {
 
 // Consume handles requests from endpoint and replies with response
 func (c *Consumer) Consume(requestPtr interface{}) (response interface{}, err error) {
-	// request := requestPtr.(*Request)
+	request, ok := requestPtr.(*Request)
+	if !ok {
+		return nil, errUnsupportedRequest
+	}
 
-	// TODO there should be some validation of the received proposal and storing it somewhere for the server needs.
-	// TODO signature validation of the promise should be here too.
+	receivedPromise, err := json.Marshal(request.SignedPromise.Promise)
+	if err != nil {
+		return nil, err
+	}
 
-	// if request.SignedPromise.IssuerSignature == valid {
-	// 		return &Response{
-	// 			Success: false,
-	// 			Message: fmt.Sprintf("Bas signature: %s", request.SignedPromise.IssuerSignature),
-	// 		}, fmt.Errorf("Bad promise signature")
-	// }
+	signature := identity.SignatureBase64(string(request.SignedPromise.IssuerSignature))
+	issuer := identity.FromAddress(request.SignedPromise.Promise.IssuerID)
+	verifier := identity.NewVerifierIdentity(issuer)
+	if !verifier.Verify(receivedPromise, signature) {
+		return failedResponse(errBadSignature, request), nil
+	}
+
+	benefiter := identity.FromAddress(request.SignedPromise.Promise.BenefiterID)
+	if benefiter.Address != c.proposal.ProviderID {
+		return failedResponse(errUnknownBenefiter, request), nil
+	}
+
+	price := c.proposal.PaymentMethod.GetPrice()
+	amount := request.SignedPromise.Promise.Amount.Amount
+	if amount < price.Amount {
+		return failedResponse(errLowAmount, request), nil
+	}
+
+	balance, err := c.etherClient.BalanceAt(context.Background(), common.HexToAddress(issuer.Address), nil)
+	if err != nil {
+		return nil, err
+	}
+	if balance.Uint64() < amount {
```

Comment:
Can `BalanceRegistry` return already parsed `money`?

Reply:
For `money.Money` we need to know `Currency` to provide it. Left only balance, since we don't know currency at that moment.